rewrite-java/src/main/java/org/openrewrite/java/ReplaceStringLiteralWithConstant.java

```diff
@@ -0,0 +1,114 @@
+        this.literalValue = literalValue;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace String Literal with Constant";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace a String Literal with named constant. If String literal value no provided, the recipe will use constant's Name in place of literal value.";
+    }
+
+    @Override
+    public JavaVisitor<ExecutionContext> getVisitor() {
+        return new JavaVisitor<ExecutionContext>() {
+
+            @Nullable
+            J.FieldAccess constant;
+
+            @Override
+            public J visitLiteral(J.Literal literal, ExecutionContext executionContext) {
+                if(isStringLiteral(literal)) {
+                    String owningType = fullyQualifiedNameOfConst.substring(0,fullyQualifiedNameOfConst.lastIndexOf("."));
+                    maybeAddImport(owningType,false);
+                    return buildConstant().withPrefix(literal.getPrefix());
+                }
+                return super.visitLiteral(literal, executionContext);
+            }
+
+            private boolean isStringLiteral(@Nullable J.Literal literal) {
+                return literal!=null && TypeUtils.isString(literal.getType()) &&
+                        ((String) literal.getValue()).equals(literalValue);
+            }
+
+            private J.FieldAccess buildConstant() {
+                if(constant == null) {
+                    //split fqn into owning type and constant name
+                    String constantName = fullyQualifiedNameOfConst.substring(fullyQualifiedNameOfConst.lastIndexOf(".")+1);
+                    String owningType = fullyQualifiedNameOfConst.substring(0,fullyQualifiedNameOfConst.lastIndexOf("."));
+                    String className = owningType.substring(owningType.lastIndexOf(".")+1);
+
+                    // Instead of creating new FieldAccess using constructor, parsing code snippet to create Field Access using JavaParser.
+                    JavaParser parser = JavaParser.fromJavaVersion().build();
+                    List<J.CompilationUnit> result = parser.parse(
+                            "import "+ owningType +";\n" +
+                                    "class Test {\n" +
+                                    "   Object o = "+String.join(".",className,constantName)+";\n" +
+                                    "}"
+                    );
+
+                    // Retrieving FieldAccess from Compilation Unit
+                    J j = ((J.VariableDeclarations) result.get(0).getClasses().get(0).getBody().getStatements().get(0)).getVariables().get(0).getInitializer();
+                    if (!(j instanceof J.FieldAccess)) {
+                        throw new IllegalArgumentException("Invalid FQN : " + fullyQualifiedNameOfConst);
+                    }
+
+                    J.FieldAccess parsedFieldAccess = (J.FieldAccess) j;
```

Comment:
I learned something new yesterday; [we now have a `PartProvider`](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/PartProvider.java), which potentially makes constructions like these a lot easier:
```java
nullableStringMethodTemplate = PartProvider.buildPart(
    "import static org.mockito.ArgumentMatchers" +
    ".nullable;\n" +
    "public class A {\n" +
    "    void method() {\n" +
    "        Object x = nullable(String.class);\n" +
    "    }\n" +
    "}", J.MethodInvocation.class, MOCKITO_CLASS_PATH);
```
Not meant to give you more work to do; If you want I can try to add this here as well. :)

Reply:
`PartProvider` was a workaround for a `JavaTemplate` bug... I'd rather we fix the `JavaTemplate` bug and get rid of `PartProvider` than use it more.